Docs update for CMDR-DOS, pinouts, and firmware update stub.

tomxp411 opened the pull request:
This is the current state of my documentation updates, with 3 new chapters. I have also updated the TOC to match.

There are several TODO items, where I just don't have enough information or have not had time to flesh out a topic. However, I felt its important to get the current information into the repo, as the emulator has features that are currently undocumented.

jestin commented on X16 Reference - 12 - Hardware.md:
```diff
@@ -0,0 +1,133 @@
+
+The IEC port is a female 6 pin DIN 45322 connector. The pinout and specifications are the same as the Commodore 128 computer, with the required lines for Fast IEC, as used by the 1571 and 1581 diskette drives. 1541 drives are also compatible, using standard IEC mode at 4800 bits/sec.
+
+<img src="images/iec_port.gif" alt-text="IEC Serial Port" />
```
Should markdown be used for images instead of <img> tags?  This could be rewritten as:
```
![IEC Serial Port](images/iec_port.gif)
```

tomxp411 replied:
As mentioned elsewhere, I have settled on using img tags exclusively because Markdown does not support the width field, and that is necessary to keep an even size on the images.

jestin commented on X16 Reference - 12 - Hardware.md:
```diff
@@ -0,0 +1,133 @@
+
+The computer contains two SNES style ports and will work with Super Nintendo compatible game pads. 
+
+<img src="images/SNES_Controller_Female.svg" alt-text="SNES Controller Port" width="256" />
```
If `width` isn't important, could this be re-written directly in markdown?
```
![SNES Controller Port](images/SNES_Controller_Female.svg)
```

tomxp411 replied:
It could be, but I’d have to re-scale the images. I will probably do that anyway, but I don’t have very good SVG fools right now.

jestin replied:
Sounds good.  I checked around to see how other images were placed and the only one I found was the VERA palette.  Hard to say that constitutes a standard.  Tags are probably fine unless someone else has a concern.

mooinglemur reviewed (commented):
The cleaned up pinout binary png file is even larger than the rougher one, and I don't think we want to include the rougher one in the commit anyway.  Binary files inside of a git repo is something I think should be minimized as much as possible, both in count and in size.

mooinglemur commented:
![expansion_slot_3](https://user-images.githubusercontent.com/395186/222121914-994f8e77-f048-4f2b-a77e-bd78eb5ff93e.png)
![expansion_slot_4](https://user-images.githubusercontent.com/395186/222121936-edf1497b-4149-4cde-9612-b1f0c90965be.png)

20k for 8 color and 5k for 2 color respectively, both IMO are plenty readable.

tomxp411 commented:
I'm planning on replacing both of those, eventually, with a hand-typed chart. For now, I have removed Matt's rough version and used Katie's cleaned-up one.

tomxp411 commented:
Removed the graphic expansion port and replaced with a table. 
Added pinouts for everything I could identify, except the User port. 

TODO: User port, TOC.

tomxp411 commented:
ok. these are about as good as I'm going to get until I have some hardware to compare against.

mooinglemur reviewed (approved):
This is all excellent work, Tom.  I'm sure a lot of people will appreciate having these much-needed reference sections!

I see a handful of typos but it seems like a good time to get this iteration merged in so that other pending updates don't start to conflict.
